torch/_inductor/lowering.py

```diff
@@ -292,10 +292,6 @@ def wrapped(*args, **kwargs):
             unpacked = True
             args = args[0]
 
-        # explicitly assert for "out=" ops for better error messages
-        assert not any(
-            x == "out" for x in kwargs.keys()
```

Comment:
We should either :

1) Generalize the fallback kernel `out` mutation handling (only handles fsdp.split_with_sizes_copy currently)

2) Still error on other ops with `kwargs=out`

Reply:
yes I think I should just generalize the fallback kernel `out` mutation handling to cover `out=Tensor` and `out=List[Tensor]` to start